cmd/sdkch/main.go

```diff
@@ -0,0 +1,277 @@
+	writeEntryFile(filename, []byte(message))
+}
+
+// nolint: errcheck
+func addEntryFile(sectionDir, stanzaDir string) {
+	bs := readUserInput()
+	firstLine := strings.TrimSpace(strings.Split(string(bs), "\n")[0])
+	filename := filepath.Join(
+		filepath.Join(entriesDir, sectionDir, stanzaDir),
+		generateFileName(firstLine),
+	)
+
+	writeEntryFile(filename, bs)
+}
+
+var filenameInvalidChars = regexp.MustCompile(`[^a-zA-Z0-9-_]`)
+
+func generateFileName(line string) string {
+	var chunks []string
+	subsWithInvalidCharsRemoved := strings.Split(filenameInvalidChars.ReplaceAllString(line, " "), " ")
+	for _, sub := range subsWithInvalidCharsRemoved {
+		sub = strings.TrimSpace(sub)
+		if len(sub) != 0 {
+			chunks = append(chunks, sub)
+		}
+	}
+
+	return strings.Join(chunks, "-")
+}
+
+func generateChangelog(version string) {
+	fmt.Printf("# %s\n\n", version)
+	for sectionDir, sectionTitle := range sections {
+
+		fmt.Printf("## %s\n\n", sectionTitle)
+		for stanzaDir, stanzaTitle := range stanzas {
+			fmt.Printf("### %s\n\n", stanzaTitle)
+			path := filepath.Join(entriesDir, sectionDir, stanzaDir)
+			files, err := ioutil.ReadDir(path)
+			if err != nil && !os.IsNotExist(err) {
+				log.Fatal(err)
+			}
+			for _, f := range files {
+				if f.Name()[0] == '.' {
+					continue
+				}
+				if err := indentAndPrintFile(filepath.Join(path, f.Name())); err != nil {
+					log.Fatal(err)
+				}
+			}
+
+			fmt.Println()
+		}
+	}
+}
+
+// nolint: errcheck
+func indentAndPrintFile(filename string) error {
+	file, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	firstLine := true
+	for scanner.Scan() {
+		line := scanner.Text()
+		if len(strings.TrimSpace(line)) == 0 {
+			continue
+		}
+
+		if firstLine {
+			fmt.Printf("* %s\n", scanner.Text())
+			firstLine = false
+			continue
+		}
+
+		fmt.Printf("  %s\n", scanner.Text())
+	}
+
+	return scanner.Err()
+}
+
+func writeEntryFile(filename string, bs []byte) {
+	if err := os.MkdirAll(filepath.Dir(filename), 0755); err != nil {
+		log.Fatal(err)
+	}
+	outFile, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE|os.O_EXCL, 0644)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer outFile.Close()
+
+	if _, err := outFile.Write(bs); err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Fprintf(os.Stderr, "Written to %s\n", filename)
+}
+
+func validateSectionStanzaDirs(sectionDir, stanzaDir string) {
+	if _, ok := sections[sectionDir]; !ok {
+		log.Fatalf("invalid section -- %s", sectionDir)
+	}
+	if _, ok := stanzas[stanzaDir]; !ok {
+		log.Fatalf("invalid stanza -- %s", stanzaDir)
+	}
+}
+
+func readUserInput() []byte {
+	bs, err := ioutil.ReadAll(os.Stdin)
+	if err != nil {
+		log.Fatalf("error: %v", err)
+	}
+	return bs
+}
+
+func printUsage() {
+	usageText := fmt.Sprintf(`usage: %s [-d DIRECTORY] COMMAND
+
+Maintain unreleased changelog entries in a modular fashion.
+
+Commands:
+    init
+    add SECTION STANZA [MESSAGE]    Add an entry file.
```

Comment:
Why? Many Unix commands help screens use all caps names for their arguments

Reply:
It's inconsistent with the SDK and cobra, also section and stanza's should be in square brackets because they are also arguments. if anything we could use `[optional-message]` if we wanted to differentiate

Reply:
Not a big deal, will amend accordingly